index.js

```diff
@@ -141,6 +141,14 @@ function printMap(val, indent, prevIndent, refs, maxDepth, currentDepth, plugins
 }
 
 function printObject(val, indent, prevIndent, refs, maxDepth, currentDepth, plugins) {
+  if (val.toJSON) {
+    while (val.toJSON) {
+      val = val.toJSON();
+    }
+    if (typeof val === 'string') {
+      return /(\\|\"|\')/.test(val) ? print(val) : val;
```

Comment:
would probably call `printString(val)` directly instead of doing the RegExp check and the `print(val)` - you already know it's a string.